tensorflow/core/kernels/cwise_op_abs.cc

```diff
@@ -38,7 +38,11 @@ REGISTER_KERNEL_BUILDER(Name("Abs")
 #endif
 
 #if TENSORFLOW_USE_SYCL
-REGISTER3(UnaryOp, SYCL, "Abs", functor::abs, float, double, int64);
+#define REGISTER_KERNEL(type) REGISTER(UnaryOp, SYCL, "Abs", functor::abs, type);
```

Comment:
Can you create a new macro that call # define REGISTER_KERNEL(type) TF_CALL_SYCL_NUMBER_TYPES(REGISTER_KERNEL) in a central location and also make sure that the new macro undefines REGISTER_KERNEL when it's done. That will avoid a lot of boilerplate code.